flocker/ca/test/test_ca.py

```diff
@@ -16,13 +16,296 @@
+        instance.
+        """
+        uc1 = UserCredential.initialize(self.path, self.ca, self.username)
+        uc2 = UserCredential.from_path(self.path, uc1.username)
+        self.assertEqual(uc1, uc2)
+
+    def test_keypair_correct_umask(self):
+        """
+        A keypair file written by ``UserCredential.initialize`` has
+        the correct permissions (0600).
+        """
+        uc = UserCredential.initialize(self.path, self.ca, self.username)
+        key_file = b"{user}.key".format(user=uc.username)
+        keyPath = self.path.child(key_file)
+        st = os.stat(keyPath.path)
+        self.assertEqual(b'0600', oct(st.st_mode & 0777))
+
+    def test_certificate_correct_permission(self):
+        """
+        A certificate file written by ``UserCredential.initialize`` has
+        the correct access mode set (0600).
+        """
+        uc = UserCredential.initialize(self.path, self.ca, self.username)
+        key_file = b"{user}.key".format(user=uc.username)
+        keyPath = self.path.child(key_file)
+        st = os.stat(keyPath.path)
+        self.assertEqual(b'0600', oct(st.st_mode & 0777))
+
+    def test_create_error_on_non_existent_path(self):
+        """
+        A ``PathError`` is raised if the path given to
+        ``UserCredential.initialize`` does not exist.
+        """
+        path = FilePath(self.mktemp())
+        e = self.assertRaises(
+            PathError, UserCredential.initialize, path, self.ca, self.username
+        )
+        expected = (b"Unable to write certificate file. "
+                    b"No such file or directory {path}").format(
+                        path=path.child("{}.crt".format(self.username)).path)
+        self.assertEqual(str(e), expected)
+
+    def test_load_error_on_non_existent_path(self):
+        """
+        A ``PathError`` is raised if the path given to
+        ``UserCredential.from_path`` does not exist.
+        """
+        path = FilePath(self.mktemp())
+        e = self.assertRaises(
+            PathError, UserCredential.from_path, path, self.username
+        )
+        expected = (b"Certificate file could not be opened. "
+                    b"No such file or directory {path}").format(
+                        path=path.child("{}.crt".format(self.username)).path)
+        self.assertEqual(str(e), expected)
+
+    def test_load_error_on_non_existent_certificate_file(self):
+        """
+        A ``PathError`` is raised if the certificate file path given to
+        ``UserCredential.from_path`` does not exist.
+        """
+        path = FilePath(self.mktemp())
+        path.makedirs()
+        cert_file = b"{user}.crt".format(user=self.username)
+        e = self.assertRaises(
+            PathError, UserCredential.from_path, path, self.username
+        )
+        expected = (b"Certificate file could not be opened. "
+                    b"No such file or directory "
+                    b"{path}").format(
```

Comment:
Exceptions stringify to the native string type (`"..."`), not necessarily to bytes.